### ISSUE TITLE
💡 [FEAT] Add a doing status and timing the task currently doing

### ISSUE THREAD

**Issue by JackTheMico** [kraanzu/dooit#173]
**Is your feature request related to a problem? 😢 Please describe.** 
A to-do task has no "starting" or "doing" status. So sometimes I would forget what I was trying to do.

**Describe the solution you'd like**   🤔
1. Add a new " doing " status for the to-do class.
2. Add a timing mechanism(a background timer) to the current task so we can know how much time we spent on tasks.
3. Need to add a field for 2. to save the spent time to the todo.yaml file.
4. Alter the TodoWidget to show the spent time.

**Describe alternatives you've considered**  🧐

**Additional context** 📝
Thanks for creating this wonderful tool, I'd like to contribute because I'm learning how to use textual right now.

**Comment by kraanzu** [comment]
Hey @JackTheMico, I see the use case and it'd be great! 

Questions:
Let's say a user starts a task.
- Will the user need to stop the task manually to record time?
- Where do you plan to show the timer?

> Thanks for creating this wonderful tool,

Thanks man!

>  I'd like to contribute because I'm learning how to use textual right now.

Sure! I can help as well after we are on the same page about the implementation :)
There can a lot of attributes related to a todo so I'm thinking of a way to let user decide how the tasks should render but haven't been able to reach a good solution yet + I really don't get much time to work on the projects to the development has been quite slow :(

**Comment by JackTheMico** [comment]
My thoughts:
1. A user should be able to pause the timer or toggle the task to the done status to complete the recording of time.
2. I would like it to be shown at the status line personally, the format would be like <the currently doing task name>--<the timer>

**Comment by kraanzu** [comment]
I see. I'll have to test some things out so lets keep it on hold for a few days and then I can take a look ;)

**Comment by Arqamz** [comment]
You could add a wakatime plugin for this also that would be great for this 'timing'

**Comment by kraanzu** [comment]
Yes I'm working on implementing plugins. Its taking time but I guess might as well take some time rather than a bad api
